sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusMessageSerializer.java

```diff
@@ -219,8 +223,25 @@ public int getSize(Message amqpMessage) {
         return Collections.emptyList();
     }
 
-    private List<Instant> deserializeListOfInstant(Message amqpMessage) {
+    private List<OffsetDateTime> deserializeListOfOffsetDateTime(Message amqpMessage) {
+        if (amqpMessage.getBody() instanceof AmqpValue) {
+            AmqpValue amqpValue = ((AmqpValue) amqpMessage.getBody());
+            if (amqpValue.getValue() instanceof  Map) {
```

Comment:
This looks very similar to the instant one, can you just do another transformation rather than duplicating this logic.

Reply:
In addition, can you flatten this logic? The multiple nested ifs loops are hard to read.
For example:
```java
if (!(amqpMessage.getBody() instanceof AmqpValue)) {
    return Collections.emptyList();
}

final AmqpValue amqpValue = ((AmqpValue) amqpMessage.getBody());
if (!(amqpValue.getValue() instanceof Map)) {
    return Collections.emptyList();
}

... stuff.

Reply:
Created a new issue to refactor all Instant related stuffs. https://github.com/Azure/azure-sdk-for-java/issues/14691